### PR TITLE
Support for Weave docker network plug-in 

### PR DESCRIPTION
**Pull request by jmcadden**
[Weave network plugin](https://github.com/weaveworks/weave) for docker provides multi-tenant overlay (VXLAN) networks that are exposed to the host. 

We support this by adding an env variable that allows the user to specify network arguments, e.g.,: 
`$ EBBRT_NODE_ALLOCATOR_DEFAULT_NETWORK_ARGUMENTS="-d weave" ./helloworld`

Because the host exposure must be done manually, I've added this logic into the NodeAllocator, and is applied only when the network argument string contains "weave".

### CODE REVIEW

File: hosted/src/NodeAllocator.cc [hunk 1]
```diff
@@ -105,14 +106,18 @@ ebbrt::NodeAllocator::NodeAllocator() : node_index_(2), allocation_index_(0) {
...
   }
   auto acceptor = std::make_shared<bai::tcp::acceptor>(
       active_context->io_service_, bai::tcp::endpoint(bai::tcp::v4(), 0));
   auto socket = std::make_shared<bai::tcp::socket>(active_context->io_service_);
 
-  auto f = popen(
-      ("docker network create " + std::to_string(std::time(nullptr))).c_str(),
-      "r");
+  auto f = popen((std::string("docker network create ") +
+                  DefaultNetworkArguments.c_str() +
```
Comment (dschatzberg) [inline hosted/src/NodeAllocator.cc]: I don't think you need c_str() on this line

**Comment by jmcadden** [comment]
@BetsyRT r+

**Comment by BetsyRT** [comment]
:pushpin: Commit 4944c28 has been approved by `jmcadden`

**Comment by BetsyRT** [comment]
:hourglass: Testing commit 4944c28 with merge 11e0ef8...

**Comment by BetsyRT** [comment]
:sunny: Test successful - [runtests](http://192.168.73.2:8010/builders/runtests/builds/44)
Approved by=jmcadden
Pushing 11e0ef870be26de680bf75267526ad2b94fde8f9 to master...

**Pull request merged**